### DISCUSSION TITLE
How to adapt the parser for different firmware (U 23B V2.3)

### DISCUSSION THREAD

**Discussion by nilsharm** [phiresky/ebv-theta-to-mqtt#4] [Q&A] [answered]
Hello, 
first things first: thank you for your excelent work!
I was able to run your scripts and dump data from my heater. However, showing a firmware version of "U 23B V2.3", the parser is not able to output reasonable data. 
![Panel](https://github.com/phiresky/ebv-theta-to-mqtt/assets/14999380/453bce86-cda1-419e-8fb1-6f5923a1e477)

Do you have a hint how to reverse engineer the raw data? Scrolling through with the naked eye is probably not the best way.

**Comment by nilsharm** [comment, marked as answer]
It looks like changing the message_prefix from b"\x21\x0**a**\x0a" to  b"\x21\x0**c**\x0a" solved my issue. 
Not sure if all values are correct but most of them seem reasonable.

**Comment by phiresky** [comment]
amazing! great to hear. no idea why you had to change that one thing though. I'll add it to the readme as a suggestion.
